### PR TITLE
UDTs completed with all the necessary variables and functions.

### PR DESCRIPTION
**Pull request by shplss**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -56,14 +56,14 @@ Purpose:  This project will show you the difference between member functions and
...
+    std::string name;                                //3
 };
 
-struct <#structName1#>                                //4
+struct C                                             //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b)                           //5
     {
         if( a->value < b->value ) return a;
```
Comment (matkatmusic) [inline main.cpp:68]: What do we always do before using a pointer?
- [ ] revise

File: main.cpp [hunk 2]
```diff
@@ -73,29 +73,37 @@ struct <#structName1#>                                //4
...
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float valFloat1{ 0 }, valFloat2{ 0 };
+    float valueUpdate(float* updVal)      //12
     {
-        
+        std::cout << "U's valFloat1 value: " << valFloat1 << std::endl;
+        valFloat1 = *updVal;
+        std::cout << "U's valFloat1 updated value: " << valFloat1 << std::endl;
```
Comment (matkatmusic) [inline main.cpp:81]: What do we always do before using a pointer?
- [ ] revise

File: main.cpp [hunk 3]
```diff
@@ -73,29 +73,37 @@ struct <#structName1#>                                //4
...
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float valueUpdate(U* that, float* updVal)        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        std::cout << "U's valFloat1 value: " << that->valFloat1 << std::endl;
+        that->valFloat1 = *updVal;
+        std::cout << "U's valFloat1 updated value: " << that->valFloat1 << std::endl;
```
Comment (matkatmusic) [inline main.cpp:97]: What do we always do before using a pointer?
- [ ] revise

File: main.cpp [hunk 4]
```diff
@@ -115,19 +123,19 @@ struct <#structname2#>
...
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T var1(5, "var1");                                             //6
+    T var2(3, "var2");                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
+    C f;                                            //7
+    auto* smaller = f.compare(&var1, &var2);                              //8
     std::cout << "the smaller one is << " << smaller->name << std::endl; //9
```
Comment (matkatmusic) [inline main.cpp:132]: What do we always do before using a pointer?
- [ ] revise

File: main.cpp (hunk 1 above)
Comment (shplss) [reply main.cpp]: we should check if it's not a nullptr.

**Pull request merged**